src/app/router.js

```diff
@@ -62,15 +64,38 @@
 
       activate();
 
+
       function activate() {
-        console.log('router Controller connected');
         self.closeSidenav();
+        if (_.isEqual(userService.user, userService.mockUser)) {
+          console.log('should execute getStorageData() only on page reload'); // require specs
+
+          getStorageData();
+          $timeout(function(){
+            getStorageData();
+            console.log('calculate executed');
+          }, 3000);
+        }
```

Comment:
the flow is simple:

on **page** open - get user data,
on calculate click - set user data

what the heck are you doing here?